File: tkpdseller/src/main/java/com/tokopedia/seller/shopscore/view/widget/SquareProgressBar.java
package com.tokopedia.seller.shopscore.view.widget;

import android.content.Context;
import android.graphics.drawable.GradientDrawable;
import android.os.Build;
import android.util.AttributeSet;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import com.akexorcist.roundcornerprogressbar.RoundCornerProgressBar;

/**
 * Created by sebastianuskh on 2/27/17.
 */
public class SquareProgressBar extends RoundCornerProgressBar {
    public SquareProgressBar(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SquareProgressBar(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void drawProgress(LinearLayout layoutProgress, float max, float progress, float totalWidth, int radius, int padding, int colorProgress, boolean isReverse) {
        GradientDrawable backgroundDrawable = createGradientDrawable(colorProgress);
        int newRadius = radius - (padding / 2);
        float rightRadius = ((progress / max) < 0.98) ? 0 : newRadius;
        backgroundDrawable.setCornerRadii(new float[]{newRadius, newRadius, rightRadius, rightRadius, rightRadius, rightRadius, newRadius, newRadius});
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
            layoutProgress.setBackground(backgroundDrawable);
        } else {
            layoutProgress.setBackgroundDrawable(backgroundDrawable);
        }

        float ratio = max / progress;
        int progressWidth = (int) ((totalWidth - (padding * 2)) / ratio);
        ViewGroup.LayoutParams progressParams = layoutProgress.getLayoutParams();
        progressParams.width = progressWidth;
        layoutProgress.setLayoutParams(progressParams);
    }
}

File: features/merchant/shop_score/src/main/java/com/tokopedia/shop/score/detail_old/view/widget/SquareProgressBar.kt
package com.tokopedia.shop.score.detail_old.view.widget

import android.content.Context
import android.os.Build
import android.util.AttributeSet
import android.widget.LinearLayout
import com.akexorcist.roundcornerprogressbar.RoundCornerProgressBar

class SquareProgressBar : RoundCornerProgressBar {
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun drawProgress(
        layoutProgress: LinearLayout,
        max: Float,
        progress: Float,
        totalWidth: Float,
        radius: Int,
        padding: Int,
        colorProgress: Int,
        isReverse: Boolean) {
        val backgroundDrawable = createGradientDrawable(colorProgress)
        val newRadius = radius - padding / 2
        val rightRadius: Float = if (progress / max < LIMIT_ROUNDED) 0f else newRadius.toFloat()
        backgroundDrawable.cornerRadii = floatArrayOf(
            newRadius.toFloat(),
            newRadius.toFloat(),
            rightRadius,
            rightRadius,
            rightRadius,
            rightRadius,
            newRadius.toFloat(),
            newRadius.toFloat())
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
            layoutProgress.background = backgroundDrawable
        } else {
            layoutProgress.setBackgroundDrawable(backgroundDrawable)
        }
        val ratio = max / progress
        val progressWidth = ((totalWidth - padding * 2) / ratio).toInt()
        val progressParams = layoutProgress.layoutParams
        progressParams.width = progressWidth
        layoutProgress.layoutParams = progressParams
    }

    companion object {
        const val LIMIT_ROUNDED = 0.98
    }
}